projects/step-frontend/src/lib/modules/_common/components/execution-link/execution-link.component.ts

```diff
@@ -0,0 +1,23 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Execution, ExecutionsService } from '@exense/step-core';
+
+@Component({
+  selector: 'step-execution-link',
+  templateUrl: './execution-link.component.html',
+  styleUrls: ['./execution-link.component.scss'],
+})
+export class ExecutionLinkComponent implements OnInit {
+  @Input() executionId?: string;
+
+  @Input() executionDescription?: string;
+
+  constructor(private _executionService: ExecutionsService) {}
+
+  ngOnInit(): void {
+    if (this.executionId && !this.executionDescription) {
```

Comment:
in fact, `ngOnInit` is fine in this case, since `executionId` will never change. It should only check in the beginning whether there is a description available and if there is none fetch it

Reply:
I'm not 100% sure how this might behave with pagination. Would pagination instantiate new objects or reuse the old once @dvladir ?